library/Booster/Pattern/Base.hs

```diff
@@ -692,9 +692,18 @@ injectionSymbol =
         }
 
 -- convenience patterns
```

Comment:
(for discussion, not on this PR)
We have grown a large amount of these convenience patterns - and they are quite specific `K` idioms.
Maybe we should separate them into their own module (or modules? `Booster.Pattern.Bool` and `Booster.Pattern.Equalities`? But where would `KSeq` go? ) and document the assumptions we make in haddock comments.

Reply:
Created the suggested `Booster.Pattern.Bool` and moved the patterns plus functions like `splitBoolPredicates` to this module